components/neeva/renderer/content_filtering_agent.cc

```diff
@@ -51,11 +51,19 @@ std::unique_ptr<base::MemoryMappedFile> MapRegion(
 ContentFilteringAgent::ContentFilteringAgent(
     blink::ThreadSafeBrowserInterfaceBrokerProxy* broker)
     : task_runner_(base::SequencedTaskRunnerHandle::Get()) {
+
+  // Binds the service as a remote and creates a pipe to the Browser.
+  // See ContentFilteringService::AddInterface() for the handling of the reciever part. 
+  // @Darin why is the Browser broker when registering the Renderer service pipe??
```

Comment:
I don't understand the question.

Reply:
Correct me if I'm wrong, I think I understand it better now: 

The browser `broker` is receiving callbacks from the `service_` here in the `agent` (which is in the renderer).

 So whenever we call `service_.onContentFiltered()` it gets sent to somewhere in the browser process. This "somewhere" that picks up the receiver is in `weblayer/browser/content_browser_client_impl.cc` where `AddInterface()` is called for each renderer process. (So the ContentFilteringService actually lives in the Browser process). 

`AddInterface()` makes the `ContentFilteringService` a self-owned receiver. 

So then in `ContentFilteringAgent`, the browser `broker` somehow grabs the right ContentFilteringService (I don't know how help) and binds the pipe to it. (which is the code we are looking at right now). 

The reason we are using the browser `broker` is because the ContentFilteringService exists and can only be accessed in the browser process. 


<img width="756" alt="Screen Shot 2022-10-26 at 11 35 41 AM" src="https://user-images.githubusercontent.com/20916043/198108342-262bb03a-42a9-44b9-ae9a-ec79b1160ac8.png">

Reply:
Offline conversation: 

In `weblayer/browser/content_browser_client_impl.cc`, the browser process is calling `neeva::ContentFilteringService::AddInterface` which creates a lambda that is ready to create the `ContentFilteringService` when a receiver is asked for.

So the life cycle is:
1. The browser process calls `ExposeInterfacesToRenderer()` which sets up the lambda to create the service (talked about above)
2. Whenever a renderer process is created, a `ContentFilteringAgent` is constructed. (see: weblayer/renderer/content_renderer_client_impl.cc) -- I found that by calling `grep -ri ContentFilteringAgent` in the `weblayer` directory
3. `ContentFilteringAgent`'s constructor calls the browser interface broker to `GetInterface()` of a receiver that is of type `mojom::ContentFilteringService`. This calls the lambda to create the Service (in the Browser Process). 
4. The agent now has a reference to a `_service` that it can call `onContentFiltered()` or `AddRulesListener()` on